packages/api/src/controllers/managementPlanController.js

```diff
@@ -173,6 +174,25 @@ const managementPlanController = {
             tasks.push(fieldWorkTask);
           }
 
+          if (req.body.assignee_user_id) {
+            await Promise.all(
+              tasks.map(async (task) => {
```

Comment:
Why `.map()` is used here? Would it be better to use `.forEach()`?
My understanding is that you want to use [.map()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map) when you want to modify a collection and want to hold to the modified values. It seems that this time we're only interested in the _side-effect_ of `sendTaskNotification`? [.forEach()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach) is better suited for this?

Reply:
Thank you, I was trying to use `Promise.all()` so that notifications would be sent at once and realized that notification was actually being sent by the side-effect... I will fix it!